src/selectors/run-status.js

```diff
@@ -0,0 +1,23 @@
+import { createSelector } from 'reselect';
+
+function groupByErrorStatus(items) {
+  const status = { success: {}, failed: {} };
+  Object.entries(items || {}).forEach(([id, item]) => {
+    if (item.error) {
+      status.failed[id] = item;
+    } else {
+      status.success[id] = item;
+    }
+  });
+  return status;
+}
+
+export const getDatasetsStatus = createSelector(
```

Comment:
this will be updated together with the https://github.com/kedro-org/kedro-viz/blob/64da498646a872475622d45e569b59d5eac4b99e/src/selectors/run-status.js once @jitu5 PR merged to the feature-branch https://github.com/kedro-org/kedro-viz/blob/64da498646a872475622d45e569b59d5eac4b99e/src/selectors/run-status.js